scripts/prune_genesis_codes.py

```diff
@@ -0,0 +1,99 @@
+import argparse
+import copy
+import json
+import os
+
+
+def load_json_file(filename):
+    """Load the entire JSON file into memory."""
+    with open(filename, 'r') as file:
+        data = json.load(file)
+    return data
+
+
+def save_json_file(filename, data):
+    """Save a dictionary to a JSON file."""
+    with open(filename, 'w') as file:
+        json.dump(data, file)
+
+
+def parse_commandline():
+    description = """This script prunes the genesis file by removing unused codes
+and updating the code IDs in the contracts. It takes the input genesis file
+and outputs the pruned genesis file.
+    """
+
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument(
+        "genesis_export", type=_path, help="The path to the genesis export"
+    )
+    parser.add_argument(
+        "--output_file",
+        help="The path to save the pruned genesis file",
+        default="out.json",
+    )
+
+    return parser.parse_args()
+
+
+def _path(text: str) -> str:
+    return os.path.abspath(text)
+
+
+def main():
+    # Parse command-line arguments
+    args = parse_commandline()
+
+    # Load genesis file
+    print("Opening genesis file")
+    genesis = load_json_file(args.genesis_export)
+
+    wasm = genesis["app_state"]["wasm"]
+    codes = wasm["codes"]
+    contracts = wasm["contracts"]
+
+    print("Building code hashes map")
+    # Create maps with code hashes and IDs
+    original_code_hash_to_code = {}
+    original_code_id_to_hash = {}
+    for code in codes:
+        original_code_hash_to_code[code["code_info"]["code_hash"]] = code
+        original_code_id_to_hash[code["code_id"]] = code["code_info"]["code_hash"]
```

Comment:
Not that it would matter much, feel fre to ignore:
```suggestion
        code_has = code["code_info"]["code_hash"]
        original_code_hash_to_code[code_hash] = code
        original_code_id_to_hash[code["code_id"]] = code[code_hash]
```